FFmpegInterop/UncompressedSampleProvider.cpp

```diff
@@ -113,17 +113,18 @@ HRESULT UncompressedSampleProvider::GetFrameFromFFmpegDecoder(AVFrame* avFrame,
 		else
 		{
 			// Update the timestamp
-			if (avFrame->pts != AV_NOPTS_VALUE)
+			if (avFrame->pts != AV_NOPTS_VALUE && avFrame->pts >= 0 && avFrame->pts - nextFramePts < 600000000)
```

Comment:
I am not 100% sure if negative pts values are always an error. Adding this >= 0 check here might introduce new problems with files out in the field. Since the problem was caused by a bug on our side, I would propose to not add >= 0 check here. For the second check, it should be changed to: (!hasNextFramePts ||  avFrame->pts - nextFramePts < 600000000). E.g. after a seek, we do not have a valid nextFramePts, so the check could cause problems on file seeks (only on files with very long durations though).